### ISSUE TITLE
Variables in deconstructors that match a deprecated property name are incorrectly reported as deprecated

### ISSUE THREAD

**Issue by ST-DDT** [gund/eslint-plugin-deprecation#75]
The second required disable comment shouldn't be needed/the lint error is reported incorrectly.

````ts
function demo(
  options: {
    /**
     * @deprecated
     */
    deprecated?: boolean;
    replacement?: boolean;
  } = {}
): boolean {
  const {
    // eslint-disable-next-line deprecation/deprecation
    deprecated = false,
    // eslint-disable-next-line deprecation/deprecation
    replacement = deprecated,
  } = options;
  return replacement;
}
````

As you can see VSCode's code highlighting shows that the second line is not deprecated/striked-through:

![grafik](https://github.com/gund/eslint-plugin-deprecation/assets/1579362/15ed187a-ca00-4120-9314-29a3f67933be)

Because it references a local variable instead of the deprecated property.

---

The first disable comment is expected/required as per https://github.com/gund/eslint-plugin-deprecation/issues/74.
